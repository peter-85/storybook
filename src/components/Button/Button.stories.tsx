import type { Meta, StoryObj } from "@storybook/react";
import { fn } from "@storybook/test";
import { Button } from "./Button";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: "Design System/Atoms/Button",
  component: Button,

  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/ttuYjxsSIoCnKgBov9iWUG/Platform_BUs?type=design&node-id=1-4&mode=design&t=gKRehrtE98Fmnv5x-0",
    },

    docs: {
      description: {
        component: "Primary UI component for user interaction",
      },
    },
  },
  tags: ["autodocs"],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes
  // Storybook is doing Automatic argType inference, but if we want to override something manually we can do it here in argTypes
  argTypes: {
    // render the prop without control
    onClick: {
      control: false,
    },
    extraStyles: {
      control: false,
    },
    text: {
      // this is shown in the documentation/description column
      description: "Button content",

      table: {
        // if we want to remove something from the table of the document and controls
        // disable: true,
        // defaultValue controls default column in the document table. if a default value is passed in the components it uses it. For example size and primary props
        defaultValue: {
          // if we want add additional info - it will be displayed in dropdown
          // detail: "details",
          summary: "button",
        },
      },
    },
    size: {
      // if we want to manipulate the control column in the table of docs and controls
      options: ["xs", "sm", "md", "lg", "xl", "2xl"],
      control: { type: "select" },
    },
    variant: {
      options: ["primary", "secondary"],
      control: { type: "radio" },
    },
  },

  args: {
    variant: "primary",
    text: "Button",
    size: "xl",
    isActive: false,
    isDisabled: false,
    onClick: fn(),
  },
} satisfies Meta<typeof Button>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    variant: "primary",
  },
};

export const Secondary: Story = {
  args: {
    ...Primary.args,
    variant: "secondary",
  },
  argTypes: {
    ...Primary.argTypes,
  },
};

export const XS: Story = {
  args: {
    size: "xs",
    ...Primary.args,
  },
};

export const SM: Story = {
  args: {
    size: "sm",
    ...Primary.args,
  },
};
export const MD: Story = {
  args: {
    size: "md",
    ...Primary.args,
  },
};
export const LG: Story = {
  args: {
    size: "lg",
    ...Primary.args,
  },
};

export const XL: Story = {
  args: {
    size: "xl",
    ...Primary.args,
  },
};

export const XXL: Story = {
  args: {
    size: "2xl",
    ...Primary.args,
  },
};
