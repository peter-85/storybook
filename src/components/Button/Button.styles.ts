import * as stylex from "@stylexjs/stylex";

export const styles = stylex.create({
  button: {
    fontFamily: "'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
    fontWeight: "700",
    border: 0,
    borderRadius: "3em",
    cursor: "pointer",
    display: "inline-block",
    lineHeight: 1,
  },
});

export const sizeStyles = stylex.create({
  small: { fontSize: "11px", padding: "8px 14px" },
  medium: { fontSize: "14px", padding: "11px 20px" },
  large: { fontSize: "16px", padding: "12px 24px" },
});

export const typeStyles = stylex.create({
  primary: { color: "white", backgroundColor: "#1ea7fd" },
  secondary: {
    color: "#339",
    backgroundColor: "transparent",
    boxShadow: "rgba(0, 0, 0, 0.15) 0px 0px 0px 1px inset",
  },
});
