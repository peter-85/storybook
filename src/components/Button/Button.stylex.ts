import { themeTokens } from "../themeTokens.stylex";
import * as stylex from "@stylexjs/stylex";

export const styles = stylex.create({
  button: {
    fontFamily: "'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
    fontWeight: "700",
    border: 0,
    borderRadius: "3em",
    cursor: "pointer",
    display: "inline-block",
    lineHeight: 1,
  },
});

export const sizeStyles = stylex.create({
  small: {
    fontSize: "11px",
    padding: `${themeTokens.spacingUnit1} ${themeTokens.spacingUnit2}`,
  },
  medium: {
    fontSize: themeTokens.fontSize6,
    padding: `${themeTokens.spacingUnit15} ${themeTokens.spacingUnit25}`,
  },
  large: {
    fontSize: themeTokens.fontSize5,
    padding: `${themeTokens.spacingUnit15} ${themeTokens.spacingUnit3}`,
  },
});

export const typeStyles = stylex.create({
  primary: { color: "white", backgroundColor: themeTokens.primary },
  secondary: {
    color: themeTokens.secondary,
    backgroundColor: "transparent",
    boxShadow: "rgba(0, 0, 0, 0.15) 0px 0px 0px 1px inset",
  },
});
