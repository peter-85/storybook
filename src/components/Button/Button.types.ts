import { Sizes, ButtonVariants } from "@src/types";
import type { StyleXStyles } from "@stylexjs/stylex";

export interface ButtonProps extends React.HTMLAttributes<HTMLDivElement> {
  text: string;
  onClick?: React.MouseEventHandler<HTMLDivElement>;
  variant?: ButtonVariants;
  size?: Sizes;
  extraStyles?: StyleXStyles;
  isActive?: boolean;
  isDisabled?: boolean;
}
