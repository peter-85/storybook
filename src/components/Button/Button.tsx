import { styles, sizeStyles, typeStyles } from "./Button.styles";
import * as stylex from "@stylexjs/stylex";

interface ButtonProps {
  label: string;
  type?: "primary" | "secondary";
  size?: "small" | "medium" | "large";
  onClick?: () => void;
}

export const Button = ({
  type = "primary",
  size = "medium",
  label,
  ...props
}: ButtonProps) => {
  return (
    <button
      type='button'
      {...stylex.props(styles.button, sizeStyles[size], typeStyles[type])}
      {...props}
    >
      {label}
    </button>
  );
};
