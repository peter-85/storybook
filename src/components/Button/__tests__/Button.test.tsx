import { render } from "@testing-library/react";
import { Button } from "../Button";

test("should match the snapshot", () => {
  const { asFragment } = render(<Button label='Button' />);

  expect(asFragment()).toMatchSnapshot();
});
