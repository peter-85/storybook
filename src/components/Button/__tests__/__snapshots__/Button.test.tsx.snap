// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <button
    class="Button__styles.button xrt6l7w x1xlr1w8 x1wty727 xgvv9zi x1ypdohk x1rg5ohu xo5v014 Button__sizeStyles.medium xif65rj x1jye8be Button__typeStyles.primary x1awj2ng x6k178u"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;
